assets/js/modules/subscribe-with-google/components/settings/SettingsEdit.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Subscribe with Google Settings Edit component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import ProgressBar from '../../../../components/ProgressBar';
+import { STORE_NAME } from '../../datastore/constants';
+import SettingsForm from './SettingsForm';
+const { useSelect } = Data;
+
+export default function SettingsEdit() {
+	const isDoingSubmitChanges = useSelect( ( select ) =>
+		select( STORE_NAME ).isDoingSubmitChanges()
+	);
+
+	let viewComponent;
+	if ( isDoingSubmitChanges ) {
+		viewComponent = <ProgressBar />;
```

Comment:
Related to the above comments about values being possibly `undefined`, you _could_ add extra checks here to also show the loading indicator if publication ID or products is `undefined`. It's not absolutely necessary, just a suggestion so that, as long as these values aren't loaded, the progress bar shows.